services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -2498,4 +2502,120 @@ export class WorkflowService {
       data: args,
     });
   }
+
+  async findDocumentById({
+    workflowId,
+    projectId,
+    documentId,
+    transaction,
+  }: {
+    workflowId: string;
+    projectId: string;
+    documentId: string;
+    transaction: PrismaTransaction | PrismaClient;
+  }) {
+    const runtimeData = await this.workflowRuntimeDataRepository.findByIdAndLock(
+      workflowId,
+      {},
+      [projectId],
+      transaction,
+    );
+    const workflowDef = await this.workflowDefinitionRepository.findById(
+      runtimeData.workflowDefinitionId,
+      {},
+      [projectId],
+      transaction,
+    );
+    const document = runtimeData?.context?.documents?.find(
+      (document: DefaultContextSchema['documents'][number]) => document.id === documentId,
+    );
+
+    const documentSchema = addPropertiesSchemaToDocument(document, workflowDef.documentsSchema);
```

Comment:
This is not documentSchema, you can call it `documentWithSchema` or just `document` - its very confusing in the rest of the code